core/src/test/resources/org/apache/calcite/test/RelOptRulesTest.xml

```diff
@@ -19738,15 +19729,9 @@ LogicalProject(A=[$0], Y=[$1])
       LogicalUnion(all=[true])
         LogicalProject(EXPR$0=[1], EXPR$1=[2])
           LogicalValues(tuples=[[{ 0 }]])
-        LogicalProject(EXPR$0=[3], EXPR$1=[null:INTEGER])
-          LogicalValues(tuples=[[{ 0 }]])
-        LogicalProject(EXPR$0=[7369], EXPR$1=[null:INTEGER])
-          LogicalValues(tuples=[[{ 0 }]])
         LogicalProject(EXPR$0=[null:INTEGER], EXPR$1=[20])
           LogicalValues(tuples=[[{ 0 }]])
-        LogicalProject(EXPR$0=[null:INTEGER], EXPR$1=[5])
-          LogicalValues(tuples=[[{ 0 }]])
-        LogicalValues(tuples=[[{ 7499, 30 }]])
+        LogicalValues(tuples=[[{ 3, null }, { 7369, null }, { 7499, 30 }, { null, 5 }]])
```

Comment:
I have a question: why aren't all the Rows merged into a single Values here, unlike other test cases? Additionally, why is the order of Rows in other Values different from before?

Reply:
I looked at the code and found that the SqlToRelConverter#convertRowValues ​​method merges RexLiteral into LogicalValues. In this case, the values ​​that cannot be converted to LogicalValues ​​are `(cast(1.1 as int), 2)` and `(null, cast(20.2 as int))`, which have type conversion.

Reply:
The order of the values ​​in LogicalValues ​​changes because the UnionToValuesRule processes them in the order of their inputs. In this case, the order of the planBefore LogicalUnion inputs changes, so the order changes after the UnionToValuesRule processes them.